modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/ImagePicker/ImagePicker.es.js

```diff
@@ -12,160 +12,275 @@
+		? ''
+		: Liferay.Language.get('add-image-description');
+
+	let {height = '0', width = '0'} = {
+		...JSON.parse(inputValue || '{}'),
+	};
+
+	height = `${height}px`;
+	width = `${width}px`;
+
+	return (
+		<>
+			<ClayForm.Group style={{marginBottom: '0.5rem'}}>
+				<input
+					id={id}
+					name={name}
+					type="hidden"
+					value={JSON.stringify(imageValues)}
+				/>
+				<ClayInput.Group>
+					<ClayInput.GroupItem className="d-none d-sm-block" prepend>
+						<ClayInput
+							className="bg-light"
+							disabled={readOnly}
+							onClick={_handleItemSelectorTriggerClick}
+							readOnly
+							type="text"
+							value={imageValues.title}
+						/>
+					</ClayInput.GroupItem>
+
+					<ClayInput.GroupItem append shrink>
+						<ClayButton
+							disabled={readOnly}
+							displayType="secondary"
+							onClick={_handleItemSelectorTriggerClick}
+							type="button"
+						>
+							{Liferay.Language.get('select')}
+						</ClayButton>
+					</ClayInput.GroupItem>
+
+					{imageValues.url && (
+						<ClayInput.GroupItem shrink>
+							<ClayButton
+								disabled={readOnly}
+								displayType="secondary"
+								onClick={_handleClearClick}
+								type="button"
+							>
+								{Liferay.Language.get('clear')}
+							</ClayButton>
+						</ClayInput.GroupItem>
+					)}
+				</ClayInput.Group>
+			</ClayForm.Group>
+
+			{imageValues.url && modalVisible && (
+				<ClayModal
+					className="image-picker-preview-modal"
+					observer={observer}
+					size="full-screen"
+					spritemap={spritemap}
+				>
+					<ClayModal.Header />
+					<ClayModal.Body>
+						<img
+							alt={imageValues.description}
+							className="d-block img-fluid mb-2 mx-auto rounded"
+							onClick={() => setModalVisible(false)}
+							src={imageValues.url}
+							style={{cursor: 'zoom-out', maxHeight: '95%'}}
+						/>
+						<p
+							className="font-weight-light text-center"
+							style={{color: '#FFFFFF'}}
+						>
+							{imageValues.description}
+						</p>
+					</ClayModal.Body>
+				</ClayModal>
+			)}
+
+			{imageValues.url && (
+				<>
+					<div className="image-picker-preview">
+						<img
+							alt={imageValues.description}
+							className="d-block img-fluid mb-2 rounded"
+							src={imageValues.url}
+						/>
+						<div
+							className="image-picker-priview-backdor"
+							onClick={() => setModalVisible(true)}
+							style={{
+								height,
+								width,
+							}}
+						>
+							<ClayIcon symbol="search" />
+						</div>
+					</div>
+
+					<ClayForm.Group>
+						<ClayInput
+							defaultValue={imageValues.description}
+							disabled={readOnly}
+							name={`${name}-description`}
+							onChange={_handleDescriptionChange}
+							placeholder={placeholder}
+							type="text"
+						/>
+					</ClayForm.Group>
+				</>
+			)}
+		</>
+	);
 };
 
-Soy.register(ImagePicker, templates);
-export {ImagePicker};
-export default ImagePicker;
+const ImagePickerProxy = connectStore(({emit, ...otherProps}) => (
```

Comment:
We need to add the `FieldBaseProxy` here

Reply:
🤦‍♂ 🤦‍♂ 🤦‍♂ 🤦‍♂ 🤦‍♂ 
I gonna update it